hoomd/pytest/test_box_resize.py

```diff
@@ -0,0 +1,128 @@
+    Args:
+        n: number of particles
+
+    Returns: absolute fractional coordinates
+
+    """
+    return np.random.uniform(-0.25, 0.25, size=(n, 3))
+
+
+@pytest.fixture(scope="module")
+def sys1(fractional_coordinates):
+    """Initial system box size and particle positions.
+    Args:
+        fractional_coordinates: Array of fractional coordinates
+
+    Returns: hoomd box object and points for the initial system
+
+    """
+    hoomd_box = hoomd.Box.from_box([1., 2., 3., 1., 2., 3.])
+    points = fractional_coordinates @ hoomd_box.matrix.T
+    return hoomd_box, points
+
+
+_box2 = [[50, 2., 3., 1., 2., 3.],  # Only change Lx
+         [1., 2., 3., 0.9, 2., 3.],   # Only change xy
+         [50., 2., 3., 0.9, 2., 3.],   # Change Lx and xy
+         [100., 200., 600., 0.9, 5., 7.]]  # Change all
+
+
+@pytest.fixture(scope='module', params=_box2)
+def sys2(request, fractional_coordinates):
+    hoomd_box = hoomd.Box.from_box(request.param)
+    points = fractional_coordinates @ hoomd_box.matrix.T
+    return hoomd_box, points
+
+
+@pytest.fixture(scope='function')
+def get_snapshot(sys1, device):
+    def make_shapshot():
+        box1, points1 = sys1
+        s = hoomd.Snapshot()
+        s.configuration.box = box1
+        s.particles.N = points1.shape[0]
+        s.particles.typeid[:] = [0] * points1.shape[0]
+        s.particles.types = ['A']
+        s.particles.position[:] = points1
+        return s
+    return make_shapshot
+
+
+_t_start = 2
+_t_a = 1
+_t_b = 1
+_t_ab = 2
+_t_ba = 1
+_variants = [
+    hoomd.variant.Power(0., 1., 0.1, _t_start, _t_start*2),
+    hoomd.variant.Ramp(0., 1., _t_start, _t_start*2),
+    hoomd.variant.Cycle(0., 1., _t_start, _t_a, _t_ab, _t_b, _t_ba)
+             ]
+
+
+@pytest.fixture(scope='function', params=_variants)
+def variant(request):
+    return request.param
+
+
+_scale_particles = [True,False]
+
+
+@pytest.fixture(scope='function', params=_scale_particles)
+def scale_particles(request):
+    return request.param
+
+
+def test_user_specified_variant(device, simulation_factory, get_snapshot,
+                                variant, sys1, sys2, scale_particles):
+    sim = hoomd.Simulation(device)
+    sim.create_state_from_snapshot(get_snapshot())
+
+    trigger = hoomd.trigger.After(variant.t_start)
+
+    box_resize = hoomd.update.BoxResize(
+        box1=sys1[0], box2=sys2[0],
+        variant=variant, trigger=trigger, scale_particles=scale_particles)
+    sim.operations.updaters.append(box_resize)
+    sim.run(variant.t_start*3 + 1)
+
+    assert box_resize.get_box(0) == sys1[0]
+    assert box_resize.get_box(variant.t_start*3) == sys2[0]
+    assert sim.state.box == sys2[0]
+    if scale_particles:
+        npt.assert_allclose(sys2[1], sim.state.snapshot.particles.position)
+    else:
+        npt.assert_allclose(sys1[1], sim.state.snapshot.particles.position)
+    if variant == _variants[2]:
+       assert box_resize.get_box(variant.t_start*3+1) == sys1[0]
```

Comment:
I think a comment explicitly mentioning this is checking that we are correctly handling the `Cycle` variant is needed. Also, this should be in the `test_get_box` test I mention above.